### ISSUE TITLE
After migrating to System.CommandLine, consider compatibility concerns with completion scripts

### ISSUE THREAD

**Issue by heaths** [dotnet/sdk#15921]
`dotnet complete` was [introduced in .NET Core 2.1](https://docs.microsoft.com/dotnet/core/tools/enable-tab-autocomplete) and, since then, many of us have been enjoying the feature in our shell profiles. For example, in PowerShell's `$profile` is might look something like this:

```powershell
Register-ArgumentCompleter -Native -CommandName dotnet -ScriptBlock {
    param($commandName, $wordToComplete, $cursorPosition)
    dotnet complete --position $cursorPosition "$wordToComplete" | ForEach-Object {
        [System.Management.Automation.CompletionResult]::new($_, $_, 'ParameterValue', $_)
    }
}
```

Meanwhile, `System.CommandLine` has always had [it's own way of doing this](https://github.com/dotnet/command-line-api/blob/main/docs/dotnet-suggest.md). What will be the migration story?

Presumably, either way will work. I would hope that `dotnet complete` would continue working, if only to call into `System.CommandLine` to do it. But should users remove it from their bash scripts? Perhaps order it *after* `System.CommandLine`'s support as a fallback if devs can't be sure all systems on which they roam (if using roaming shell scripts) will have net6.0 or newer?

Or could .NET even improve upon this with existing `dotnet complete` support? For example, [the first line](https://github.com/dotnet/command-line-api/blob/3e11e14238411495eb3460a35056803c31a5c061/src/System.CommandLine.Suggest/dotnet-suggest-shim.ps1#L2) of the `System.CommandLine` completion script is pretty expensive for shell start up. Perhaps it could deferred to the first attempted use, but eventually devs are going to pay the cost of a disk search. Is there an opportunity here to avoid that cost?

**Comment by sfoslund** [comment]
>  I would hope that dotnet complete would continue working, if only to call into System.CommandLine to do it.

Yes, dotnet complete is currently still supported after the move to system.commandline in .NET 6 and it is implemented using system.commandline now. 

@KathleenDollard do you want to comment on long term plans for dotnet complete/suggest?
